### ISSUE TITLE
wininput: mouse wheel

### ISSUE THREAD

**Issue by GralfR** [cbdevnet/midimonster#65]
I didn't find a support for the mouse-wheel in the wininput backend. Would be great, if this was supported.
My goal: use a midi-controller as a mouse on windows. Mouse-positioning is done by two midi control-changes, the mouse-wheel is turned by another control-change-event.
This makes scrolling documents easier and even allows using MA2 encoders with onPC.

**Comment by cbdevnet** [comment]
Hey, cool idea! :D 

One problem with the mouse wheel is that it is inherently a relative channel (it doesn't output an absolute *position* value, but changes in the position). Full support for mapping these channels in a proper way is an open ticket for post-1.0 (#41).

I will probably add basic support for the mouse wheel to the backend a way similar to the relative axes in the `evdev` backend, which means users will need to configure a range in which the wheel's relative values will be applied, which can then be used as an absolute channel.

**Comment by GralfR** [comment]
No problem with relative inputs. Some controllers like the BCF2000 (and I think also the BCR2000) which have rotary encoders offer some kind of relative mode for the encoders. The encoder remains in some center position like 50%. If it is turned left, a value less than 50% is send. If the encoder is turned right, a value more than 50% is send. The distance from the default 50% is defined by the speed, the encoder is turned.
Maybe this could inspire some ideas...

**Comment by cbdevnet** [comment]
Yes, there are several different ways to represent these relative changes as absolute values (as is often done with MIDI controllers), but the problem I'm referring to is that currently, the MIDIMonster works on the mathematical basis of being able to normalize and denormalize channels. Normalization as a mathematical concept only makes sense on absolute values (where you have a reference/range against which to normalize). Since that's not possible with relative values directly, workarounds are in place until I have figured out how to properly handle this for all cases.

Implementing one specific way of encoding relative changes (such as the BCF one) unfortunately still leaves out all other backends.
